sycl/include/CL/sycl/detail/pi.h

```diff
@@ -66,6 +66,12 @@
 #define _PI_H_VERSION_STRING                                                   \
   _PI_CONCAT(_PI_H_VERSION_MAJOR, _PI_H_VERSION_MINOR)
 
+// _PI_VERSION_MAJOR_STRING_LEN will be used to check the major versions of
+// plugins versus the major versions of PI.
+#define _PI_STRING_SUBSTITUTE(X) _PI_STRING_HELPER(X)
+#define _PI_VERSION_MAJOR_STRING_LEN                                           \
+  sizeof(_PI_STRING_SUBSTITUTE(_PI_H_VERSION_MAJOR))
+
```

Comment:
Sorry for not being clear, but I meant to implement that check here, like

#define _PI_PLUGIN_VERSION_CHECK(PiApiVersion, PiPluginVersion) \
strcmp...

or even better inline function. And call that from each plugin already, not only CUDA

Reply:
Sure. Change made. Using macro function for consistency with other macro fctns